feat(pulsar): support subscribeInitialPosition on Pulsar consumer

ericsyh opened the pull request:
# Description

`SubscriptionInitialPosition` is the start consuming position for Pulsar subscription. Similar to Apache Kafka's `initialOffset`.
* https://pkg.go.dev/github.com/apache/pulsar-client-go/pulsar#SubscriptionInitialPosition
* https://v1-12.docs.dapr.io/reference/components-reference/supported-pubsub/setup-apache-kafka/

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation / Created PR in https://github.com/dapr/docs/pull/4582

JoshVanL reviewed (changes requested):
Thanks @ericsyh! Please could you open up a PR in the [docs](https://github.com/dapr/docs/blob/v1.15/daprdocs/content/en/reference/components-reference/supported-pubsub/setup-pulsar.md) for this new field?

ericsyh commented:
> Thanks @ericsyh! Please could you open up a PR in the [docs](https://github.com/dapr/docs/blob/v1.15/daprdocs/content/en/reference/components-reference/supported-pubsub/setup-pulsar.md) for this new field?

@JoshVanL Sure, i created the PR for doc update: https://github.com/dapr/docs/pull/4582

ericsyh commented:
@JoshVanL Hey Josh, Could you help give a reivew on this PR?

cicoyle commented on pubsub/pulsar/pulsar.go:
```diff
@@ -421,6 +431,30 @@ func getSubscribeType(subsTypeStr string) pulsar.SubscriptionType {
+
+	switch subsPositionStr {
+	case subscribePositionEarliest:
+		subsPosition = pulsar.SubscriptionPositionLatest
```
```suggestion
		subsPosition = pulsar. subscribePositionEarliest
```
Should it be this?

ericsyh replied:
yes, good catch.

cicoyle commented on pubsub/pulsar/pulsar_test.go:
```diff
@@ -117,6 +117,58 @@ func TestParsePulsarMetadataSubscriptionType(t *testing.T) {
+			}
+
+			require.NoError(t, err)
+			assert.Equal(t, tc.expected, meta.SubscriptionType)
```
Should `SubscriptionType` be `SubscriptionInitialPosition ` instead? 
```suggestion
			assert.Equal(t, tc.expected, meta.SubscriptionInitialPosition)
```

ericsyh replied:
yeap, should be `SubscriptionInitialPosition` here.

ericsyh commented:
> Hey @ericsyh thanks for the PR 🎉 - mind running the linter locally and resolving the issues?

@cicoyle Hi Cassie, thx for your review and good catches. I just fixed the tests and run the lint locally. At least this PR doesn't introduce new lint issue from the output. 

```make lint | grep pulsar
WARN The linter 'gomnd' is deprecated (since v1.58.0) due to: The linter has been renamed. Replaced by mnd.
WARN The linter 'execinquery' is deprecated (since v1.58.0) due to: The repository of the linter has been archived by the owner.
WARN [runner/nolint] Found unknown linters in //nolint directives: usetesting
pubsub/pulsar/pulsar.go:25:2                                                         typecheck  could not import sync/atomic (-: could not load export data: internal error in importing "sync/atomic" (unsupported version: 2); please report an issue)
pubsub/pulsar/pulsar.go:362:27                                                       typecheck  undefined: avro
pubsub/pulsar/pulsar.go:367:9                                                        typecheck  undefined: avro
make: *** [lint] Error 1
```

ericsyh commented:
@cicoyle @JoshVanL @yaron2 hey guys, could you help take a look this PR? Since this PR is pending for review and the related doc PR https://github.com/dapr/docs/pull/4582 was closed for stale.

ericsyh commented:
> > @cicoyle @JoshVanL @yaron2 hey guys, could you help take a look this PR? Since this PR is pending for review and the related doc PR [dapr/docs#4582](https://github.com/dapr/docs/pull/4582) was closed for stale.
> 
> Yes, apologies for the slow feedback. Workflows now running

Sure that's fine. The CI workflow check passed.

marcduiker commented:
@holopin-bot @ericsyh  Thank you! Here's a digital badge as a small token of appreciation.
